src/bpm/config/job_config.go

```diff
@@ -130,6 +130,40 @@ func (c *ProcessConfig) Validate(boshRoot string, defaultVolumes []string) error
 	return nil
 }
 
+func (c *ProcessConfig) AddVolumes(volumes []string) error {
+	for _, volume := range volumes {
+		fields := strings.Split(volume, ":")
+
+		if len(fields) > 2 {
+			return fmt.Errorf("invalid volume definition: %s", volume)
+		}
+
+		v := Volume{
+			Path: fields[0],
+		}
+
+		if len(fields) == 2 {
+			options := strings.Split(fields[1], ",")
+			for _, option := range options {
+				switch option {
+				case "writable":
+					v.Writable = true
+				case "mount_only":
+					v.MountOnly = true
+				case "allow_executions":
+					v.AllowExecutions = true
```

Comment:
I wonder if we can apply these by reading the struct tags on the volume object. It may reduce the maintenance in the future?

Reply:
Definitely not essential.

Reply:
For the time being I'm gonna hold off on this one because I am not a huge fan of the `reflect` package. If we see this diverging or being hard to maintain we can switch it.